src/pm/hydra/mpiexec/mpiexec.c

```diff
@@ -222,6 +222,12 @@ int main(int argc, char **argv)
         HYDU_ERR_POP(status, "unable to create proxy list\n");
     }
 
+    if (pg->proxy_count == 1 && !HYD_server_info.user_global.launcher &&
+        !HYD_server_info.user_global.skip_launch_node) {
+        /* reset launcher to fork for local only launch */
+        HYDT_bsci_info.launcher = "fork";
```

Comment:
Does this affect spawn?

I think we should move the logic to lower layer at each proxy level. We should be able to tell which proxy is a local proxy.

Reply:
After offline chat and further inspection of the code, the "common" launcher used by both `ssh` and `fork` is supposed to have an "autofork" feature to do exactly what we want. Need to find out why it doesn't work on LCRC Swing.

Reply:
In the Swing case, localhost detection for autofork fails because of inconsistent use of FQDN. The pbs nodefile contains fully quailified hostnames e.g. `gpu3.lcrc.anl.gov` while `gethostname` returns `gpu3` and nothing else.